data/management/commands/run_schedules.py

```diff
@@ -114,28 +114,42 @@ def run_and_update_schedule(self, schedule, runInstance):
 
             # This makes sure the next runtime is calculated from the time the last one was run i.e current time
             updated_runtime = self.update_next_runtime_to_previous_start_time(schedule, runInstance.start_at, make_aware(datetime.now()))
-            self.create_next_run_instance(schedule, updated_runtime)
+            actual_runtime = self.is_updated_runtime_less_than_expected_runtime(updated_runtime, runInstance, schedule)
+            self.create_next_run_instance(schedule, actual_runtime)
         except:
             self.update_run_instance(runInstance, 'e', 'An unexpected error occured while executing the script ... please contact the administrator')
             updated_runtime = self.update_next_runtime_to_previous_start_time(schedule, runInstance.start_at, make_aware(datetime.now()))
-            self.create_next_run_instance(schedule, updated_runtime)
+            actual_runtime = self.is_updated_runtime_less_than_expected_runtime(updated_runtime, runInstance, schedule)
+            self.create_next_run_instance(schedule, actual_runtime)
 
     def update_next_runtime_to_previous_start_time(self, schedule, instance_start_date, current_date_time):
         next_runtime = self.calculate_next_runtime(
             current_date_time,
             schedule.interval,
             schedule.interval_type
         )
-        if schedule.interval_type in ('sec', 'min', 'hrs'):
-            return next_runtime
-        else:
-            updated_runtime = next_runtime.replace(
-                hour=instance_start_date.hour,
-                minute=instance_start_date.minute,
-                second =instance_start_date.second
+        updated_runtime = next_runtime.replace(
+            hour=instance_start_date.hour,
+            minute=instance_start_date.minute,
+            second =instance_start_date.second
+        )
+        return updated_runtime
+
+    def is_updated_runtime_less_than_expected_runtime(self, updated_runtime, current_runInstance, schedule):
```

Comment:
This naming implies that this returns a boolean...

Reply:
Updated function naming